binderhub-service/templates/docker-api-network-policy.yaml

```diff
@@ -0,0 +1,34 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: binder-dind-network-policy
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: binderhub-service
+      app.kubernetes.io/component: docker-api
+  policyTypes:
+    - Egress
+  egress:
+    # Allow access to everything except private IPs
+    - to:
+        - ipBlock:
+            cidr: 0.0.0.0/0
+            except:
+              - 10.0.0.0/8
+              - 172.16.0.0/12
+              - 192.168.0.0/16
+    # Allow access to DNS
+    - ports:
+        - port: 53
+          protocol: UDP
+        - port: 53
+          protocol: TCP
+      to:
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: kube-system
+          podSelector:
+            matchLabels:
+              # This label is also added to EKS core-dns deployments
```

Comment:
Can you validate what GKE and AKS does as well?

Reply:
I just checked. All good!